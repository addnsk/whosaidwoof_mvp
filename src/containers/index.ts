export * from "./Header";
export * from "./Theme";
export * from "./SelectList";
