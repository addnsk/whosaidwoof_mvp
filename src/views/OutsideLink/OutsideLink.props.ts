import * as React from "react";

export interface IOutsideLinkProps
  extends React.DetailedHTMLProps<
    React.LinkHTMLAttributes<HTMLLinkElement>,
    HTMLLinkElement
  > {
  caption: string;
}
