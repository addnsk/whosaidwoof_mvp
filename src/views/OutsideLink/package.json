{
  "name": "outside-link-view",
  "version": "0.1.0",
  "private": true,
  "main": "OutsideLink.tsx"
}
