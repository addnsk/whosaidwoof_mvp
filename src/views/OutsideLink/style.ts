import styled from "styled-components";

export const Link = styled.a`
  color: ${({ theme }) => theme.colors.light};

  outline: none;
  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);

  &:focus {
    box-shadow: 0 0 0 1px ${({ theme }) => theme.colors.accentLight};
  }
`;
