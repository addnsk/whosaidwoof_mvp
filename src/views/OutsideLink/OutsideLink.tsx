import * as React from "react";
import { IOutsideLinkProps } from "./OutsideLink.props";
import { Link } from "./style";

export const OutsideLink: React.FC<IOutsideLinkProps> = ({
  href,
  caption,
}): JSX.Element => {
  return (
    <Link href={href} target="_blank">
      {caption}
    </Link>
  );
};
