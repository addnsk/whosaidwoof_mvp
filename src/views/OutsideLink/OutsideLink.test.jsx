import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import { Theme } from "containers";
import { OutsideLink } from "./OutsideLink";

describe("OutsideLink component", () => {
  it("OutsideLink renders with full props", () => {
    render(
      <Theme>
        <OutsideLink href="/" caption="Hello" />
      </Theme>
    );
    expect(screen.getByRole("link")).toBeInTheDocument();
    expect(screen.getByRole("link")).toHaveStyle({
      color: "#FFFFFF",
    });

    expect(screen.getByText(/hello/i)).toBeInTheDocument();
    expect(screen.getByText(/hello/i).tagName).toEqual("A");

    expect(screen.getByRole("link")).toMatchSnapshot();
  });
});
