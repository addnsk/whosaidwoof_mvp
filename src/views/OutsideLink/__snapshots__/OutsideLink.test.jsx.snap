// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OutsideLink component OutsideLink renders with full props 1`] = `
<a
  class="sc-iBPTik cHDhQU"
  href="/"
  target="_blank"
>
  Hello
</a>
`;
